### PR TITLE
advective form of the GM skew diffusivity

### PR DESCRIPTION
**Pull request by simone-silvestri**
This PR implements a GM closure in the advective form. The scheme seems to work nicely

The output of the `test_gm_advection.jl`, which is basically the `coarse_baroclinic_adjustment.jl` validation case without Redi component and only with advective GM using WENO, running 300 days:

https://github.com/user-attachments/assets/172463b0-1f16-451f-8dc4-8e4a6327214a

### CODE REVIEW

**Comment by simone-silvestri** [comment]
The results of the same case performed with an immersed boundary comparing diffusive and advective formulation with same diffusivity and tapering parameters. Top, diffusive formulation, bottom advective formulation

https://github.com/user-attachments/assets/8e829290-7052-4b35-95f6-eef8d13369b4

https://github.com/user-attachments/assets/2a763f81-b517-4ef0-8dbe-98ef7b01ce13

The diffusive formulation struggles near the immersed boundaries leading to significant tracer dispersion in the backward-facing step. The advective formulation has no such problems

**Comment by glwagner** [comment]
@simone-silvestri can you instead implement this as an option within `IsopycnalSkewSymmetricDiffusivity`? I think this is nicer. One can switch between the two implementations easily then.

File: src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl [hunk 1]
```diff
@@ -60,7 +60,9 @@ IsopycnalSkewSymmetricDiffusivity(FT::DataType; kw...) =
     IsopycnalSkewSymmetricDiffusivity(VerticallyImplicitTimeDiscretization(), FT; kw...)
 
 function with_tracers(tracers, closure::ISSD{TD, A, N}) where {TD, N}
-    κ_skew = !isa(closure.κ_skew, NamedTuple) ? closure.κ_skew : tracer_diffusivities(tracers, closure.κ_skew)
+    # For the moment, allow only one skew coefficient for all tracers
+    # TODO: maybe generalize it?
+    κ_skew = closure.κ_skew 
     κ_symmetric = !isa(closure.κ_symmetric, NamedTuple) ? closure.κ_symmetric : tracer_diffusivities(tracers, closure.κ_symmetric)
```
Comment (glwagner) [inline src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl:68]: doesn't `tracer_diffusivities` already handle the `NamedTuple` case?

Comment (simone-silvestri) [reply src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl:78]: Ah I have already changed it 😅

Comment (glwagner) [reply src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl:78]: word

File: src/Models/HydrostaticFreeSurfaceModels/prescribed_hydrostatic_velocity_fields.jl [hunk 2]
```diff
@@ -87,6 +87,15 @@ end
 @inline fill_halo_regions!(::FunctionField, args...) = nothing
 
 @inline datatuple(obj::PrescribedVelocityFields) = (; u = datatuple(obj.u), v = datatuple(obj.v), w = datatuple(obj.w))
+@inline velocities(obj::PrescribedVelocityFields) = (u = obj.u, v = obj.v, w = obj.w)
+
+# Extend sum_of_velocities for `PrescribedVelocityFields`
+@inline sum_of_velocities(U::PrescribedVelocityFields, V) = sum_of_velocities(velocities(U), V)
```
Comment (glwagner) [inline src/Models/HydrostaticFreeSurfaceModels/prescribed_hydrostatic_velocity_fields.jl:93]: its more like `u1, u2, u3` right?

File: src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl [hunk 3]
```diff
@@ -179,6 +210,10 @@ end
 @inline get_tracer_κ(κ::NamedTuple, tracer_index) = @inbounds κ[tracer_index]
 @inline get_tracer_κ(κ, tracer_index) = κ
 
+# Remove skew coefficient if we are using the advective formulation
+@inline skew_coefficient(i, j, k, grid, closure, κ, args...) = κ(i, j, k, grid, args...)
+@inline skew_coefficient(i, j, k, grid, ::AdvectiveSkewClosure, args...) = zero(grid)
```
Comment (glwagner) [inline src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl:215]:
```suggestion
@inline skew_diffusivity(i, j, k, grid, ::AdvectiveSkewClosure, args...) = zero(grid)
```

perhaps?

File: src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl [hunk 4]
```diff
@@ -1,60 +1,82 @@
...
-calculated isopycnal slope values.
+calculated isopycnal slope values. The skew fluxes can be computed using either the `AdvectiveFormulation` 
+or the `DiffusiveFormulation`.
     
 Both `κ_skew` and `κ_symmetric` may be constants, arrays, fields, or functions of `(x, y, z, t)`.
 """
 function IsopycnalSkewSymmetricDiffusivity(time_disc::TD = VerticallyImplicitTimeDiscretization(), FT = Float64;
                                            κ_skew = 0,
                                            κ_symmetric = 0,
+                                           skew_fluxes_formulation::A = AdvectiveFormulation(),
```
Comment (glwagner) [inline src/TurbulenceClosures/turbulence_closure_implementations/isopycnal_skew_symmetric_diffusivity.jl:47]:
```suggestion
                                           skew_flux_formulation::A = AdvectiveFormulation(),
```

**Review by glwagner — APPROVED** [review]
Excited to use this

**Pull request merged**